### PR TITLE
[inductor] Consolidate constant_args and cpp_constant_args

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98742
* #98741

Summary: Refactor code to simplify the logic. Support convolution as an
extern call in CudaWrapperCodeGen.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

Differential Revision: [D44957788](https://our.internmc.facebook.com/intern/diff/D44957788)

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -2677,30 +2677,38 @@ def apply_constraint(self):
...
             if V.graph.cpp_wrapper:
-                for arg_name in self.ordered_kwargs_for_cpp_kernel:
-                    assert arg_name in self.kwargs, (
-                        "arg %s not found in self.kwargs" % arg_name
-                    )
-                    v = self.kwargs.get(arg_name)
-                    kwargs.append(repr(v))
+                if self.kwargs:
+                    # TODO: use native_functions.yaml as the ground truth
+                    assert (
```
Comment (desertfire) [inline torch/_inductor/ir.py:2694]: This should partly address your comment at https://github.com/pytorch/pytorch/pull/98741#discussion_r1162498456

**Review by jansel — APPROVED** [review]

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D44957788).

**Pull request closed without merging**